scripts/superdesk/datetime/datetime.js

```diff
@@ -150,11 +150,12 @@ export default angular.module('superdesk.datetime', [
 
     .directive('sdDatetime', DateTimeDirective)
 
-    .filter('reldate', function reldateFactory() {
+    .filter('reldate', ['gettextCatalog', function reldateFactory(gettextCatalog) {
         return function reldate(date) {
+            moment.locale(gettextCatalog.currentLanguage);
```

Comment:
you are right, moved to
https://github.com/superdesk/superdesk-client-core/pull/946/files#diff-72e8716fc4deadd7a7a6fa4e986cf46eL22

so it reacts as soon as the language is loaded/changed